cloudinit/config/cc_ubuntu_advantage.py

```diff
@@ -208,20 +213,36 @@ def configure_ua(token=None, enable=None, config=None):
             error=str(e)
         )
         util.logexc(LOG, msg)
-        raise RuntimeError(msg) from e
+
+        if not re.search("This machine is already attached to", str(e)):
```

Comment:
Instead of relaying on the error message, we could allow the status code `2` in `subp.subp`. That have the added benefit of not logging an error message in this case, which could confuse the user.

See the main message, containing a diff proposal.